common/scala/src/main/scala/whisk/core/database/MultipleReadersSingleWriterCache.scala

```diff
@@ -97,7 +99,7 @@ trait MultipleReadersSingleWriterCache[W, Winfo] {
     protected val cacheEnabled = true
 
     /** Subclasses: tell me what key to use for updates. */
-    protected def cacheKeyForUpdate(w: W): Any
+    protected def cacheKeyForUpdate(w: W): CacheKey
```

Comment:
with the introduction of `CacheKey` you may be able to just nuke this method (if you introduce constructors for the various types as suggested below).